packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.h

```diff
@@ -19,6 +19,18 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+#define RCTComponentViewShouldBeRecycled(recycled)    \
+  static bool _shouldBeRecycled = recycled;           \
+  +(bool)shouldBeRecycled                             \
+  {                                                   \
+    return _shouldBeRecycled;                         \
+  }                                                   \
+                                                      \
+  +(void)setShouldBeRecycled : (bool)shouldBeRecycled \
+  {                                                   \
+    _shouldBeRecycled = shouldBeRecycled;             \
+  }
```

Comment:
I don't think this is the right approach because this is a static method. It means that it applies to ALL the components of a given type.
So, let's imagine that we set `RCTComponentViewShouldBeRecycled(false)` to `RCTView`, we are disabling the recyclability for ALL the `RCTViews`. 
The possibility to be recycled or not should be a choice made instance per instance. 

I should be able to say that my `RCTImage x` should not be recycled (perhaps it is a large image and I want to keep it i memory as a perf optimization, while my `RCTImage y` should not.

What do you think?

cc. @sammy-SC which knows this part of the Framework better than me.

Reply:
Yeah, I just follow the custom user components because it's class method implemented in https://github.com/facebook/react-native/blob/ebb55a780a31ee985f48536b3eb7794cdff03304/packages/react-native/React/Fabric/Mounting/RCTComponentViewFactory.mm#L99

Maybe we can expose some like a delegate to `AppDelegate` and give its instance to ask the user to decide whether to disable recycle?